### PR TITLE
ceph-rest-api: Various REST API fixes

### PR DESCRIPTION
**Pull request by wido**
This PR fixes/changes a few things to the (legacy?) REST API. This API is still being used though by people!

- Set default values
- Listen on IPv6 addresses
- Make it easy to run under WSGI

### CODE REVIEW

File: src/pybind/ceph_rest_api.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # Globals and defaults
 #
 
-DEFAULT_ADDR = '0.0.0.0'
+DEFAULT_ADDR = '::'
```
Comment (tchaikov) [inline src/pybind/ceph_rest_api.py]: please note, this is not a Linux specific behavior. werkzeug does not `setsockopt(sock, IPPROTO_IPV6, IPV6_V6ONLY)`, neither does the builtin `SocketServer` set this socket option. that's why we have a dual-stack socket when the app is bound to "::".

**Comment by liewegas** [comment]
@wido BTW would turning this into a mgr module work for your environment?  (We'd welcome patches that do that! :)

**Comment by wido** [comment]
@liewegas: Yes, it probably will!

The API is used in a couple of environments I know for just querying data. The new restful module of the mgr will probably work longer term, but I don't think a lot of people would like to rewrite all their code right now.

As this code is present in the current Luminous RC I just made this PR to keep the things working for now.

A backport of these fixes to Luminous would be very welcome.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-27_05:44:05-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**